api/organisations/chargebee/webhook_handlers.py

```diff
@@ -0,0 +1,86 @@
+import logging
+
+from rest_framework import status
+from rest_framework.exceptions import ValidationError
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from organisations.models import Subscription
+from organisations.subscriptions.constants import (
+    SUBSCRIPTION_BILLING_STATUS_ACTIVE,
+    SUBSCRIPTION_BILLING_STATUS_DUNNING,
+)
+
+from .serializers import PaymentFailedSerializer, PaymentSucceededSerializer
+
+logger = logging.getLogger(__name__)
+
+
+def payment_failed(request: Request) -> Response:
+    serializer = PaymentFailedSerializer(data=request.data)
+
+    try:
+        serializer.is_valid(raise_exception=True)
+    except ValidationError:
+        logger.warning(
+            "Serializer failure during chargebee payment failed processing",
+            exc_info=True,
+        )
```

Comment:
Given our current conversations about logging visibility, I'm keen for this to fail loudly. We could either change this to `logger.error` (probably the best option), or perhaps we can just remove the `try` / `except`.

Reply:
Well, we could issue an error here, but it's not really an error in every case if we ever get payments that aren't associated with a subscription or if the dunning hasn't come into effect. We certainly don't want to remove the `try` / `except` because we want to continue to mark the webhook as processed, as a non `2xx` HTTP status code will have chargebee continue to retry the webhook over a short time.

Reply:
Ok, makes sense.